stairway/src/main/java/bio/terra/stairway/queue/QueueMessageReady.java

```diff
@@ -37,11 +46,13 @@ public boolean process(StairwayImpl stairwayImpl) throws InterruptedException {
               + stairwayImpl.getStairwayName()
               + (resumed ? " resumed flight: " : " did not find flight to resume: ")
               + flightId);
-      return true;
```

Comment:
Was this change necessary? If it works either way, I think returning a boolean directly is clearer than creating state which must be returned at the end.

Reply:
Because I added a `finally` clause to restore the thread's context to its original state, I do think this was a necessary change.

Reply:
Ah, but it's possible that this won't be necessary if I make a change like you described here -- https://github.com/DataBiosphere/stairway/pull/152/files/93f139fd72b2e1d0b82000676a5a163b8130236e#r1676240309

I'll take a closer look.

Reply:
> Because I added a finally clause to restore the thread's context to its original state, I do think this was a necessary change.

Even if you `return` inside a `try`, the `finally` will still run.

Reply:
I reverted to the old return strategy.